app/controllers/contacts_controller.rb

```diff
@@ -1,37 +1,33 @@
 class ContactsController < ApplicationController
-  before_filter :authenticate_user!
+  before_action :authenticate_user!
+  before_action :set_subject
 
   def new
     @message = Message.new
-    @subject = default_subject
-    user = User.find_by_id(params[:user_id])
-    if user.nil?
-      redirect_to root_path and return
-    end
-    @user = user
+    @user = User.find params[:user_id]
+    redirect_to root_path if @user.nil?
   end
 
   def create
-    @user = User.find_by_id(params[:user_id])
+    @user = User.find params[:user_id]
     @message = Message.new({
         :body => params[:body].html_safe,
-        :subject => default_subject,
+        :subject => @subject,
```

Comment:
Ah, looks like I missed this! Have fixed this in another commit on this PR.
Have used an instance variable since the new method assigns the message text to @message,
    @message = default_subject.